### PR TITLE
Feature 264 system wide scripts

### LINKED ISSUE #264: System wide scripts

### PR DESCRIPTION
**Pull request by platonovsimeon**
Resolves #264

### CODE REVIEW

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+/**
+ * Tests for Scripts functionality
+ */
+@QuarkusTest
+@QuarkusTestResource.List(
+  QuarkusTestResource(MysqlResource::class),
+  QuarkusTestResource(MetaformKeycloakResource::class)
+)
+@TestProfile(GeneralTestProfile::class)
+class ScriptsTestsIT {
```
Comment (anttileppa) [inline src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt:25]: Tests with test builder can also be simplified by using the single-expression syntax.

e.g.

fun testCreateScript() {

-->
fun testCreateScript() = TestBuilder().use { builder ->

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+      builder.assertApiCallFailStatus(403) { builder.anon.scripts.update(createdScript) }
+      builder.assertApiCallFailStatus(403) { builder.test1.scripts.update(createdScript) }
+      builder.assertApiCallFailStatus(403) { managerAuthentication.scripts.update(createdScript) }
+      builder.assertApiCallFailStatus(403) { adminAuthentication.scripts.update(createdScript) }
+    }
+  }
+
+  @Test
+  fun testFindScriptPermission() {
+    TestBuilder().use { builder ->
```
Comment (anttileppa) [inline src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt:197]: This does not tests for metaform admin permission

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+      builder.assertApiCallFailStatus(403) { builder.anon.scripts.create(script) }
+      builder.assertApiCallFailStatus(403) { builder.test1.scripts.create(script) }
+      builder.assertApiCallFailStatus(403) { managerAuthentication.scripts.create(script) }
+      builder.assertApiCallFailStatus(403) { adminAuthentication.scripts.create(script) }
+    }
+  }
+
+  @Test
+  fun testUpdateScriptPermission() {
+    TestBuilder().use { builder ->
```
Comment (anttileppa) [inline src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt:173]: This does not tests for metaform admin permission

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt (hunk 3 above)
Comment (platonovsimeon) [reply src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt]: Could you tell more specifically? It already has builder.assertApiCallFailStatus(403) { adminAuthentication.scripts.update(createdScript) }

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt (hunk 2 above)
Comment (platonovsimeon) [reply src/test/java/fi/metatavu/metaform/server/test/functional/tests/ScriptsTestsIT.kt]: Could you tell more specifically? It already has builder.assertApiCallFailStatus(200) { adminAuthentication.scripts.update(createdScript) }

File: src/main/kotlin/fi/metatavu/metaform/server/controllers/MetaformScriptController.kt [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   *
+   * @param metaform metaform
+   * @param script script
+   * @param creatorId creator id
+   */
+  fun createMetaformScript(metaform: Metaform, script: Script, creatorId: UUID) {
+    metaformScriptDAO.createMetaformScript(id = UUID.randomUUID(), metaform = metaform, script = script, creatorId = creatorId)
+  }
+
+  fun deleteMetaformScriptsByMetaform (metaform: Metaform) {
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/controllers/MetaformScriptController.kt:32]: Add missing docs

File: src/main/kotlin/fi/metatavu/metaform/server/controllers/MetaformScriptController.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    metaformScriptDAO.listByMetaform(metaform).forEach { metaformScript ->
+      metaformScriptDAO.delete(metaformScript)
+    }
+  }
+
+  fun updateMetaformScripts(metaform: fi.metatavu.metaform.api.spec.model.Metaform, updatedMetaform: Metaform, creatorId: UUID) {
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/metaform/server/controllers/MetaformScriptController.kt:38]: Add missing docs

**Review by anttileppa — APPROVED** [review]

**Pull request merged**